libs/utils/env.py

```diff
@@ -474,19 +485,36 @@ def _init_target(self, force = False):
             # Brackets are there to let the output dir be created automatically
                                     '--workload-automation-vio={}'])
 
+            # Add variant suffix
+            suffix = os.path.splitext(os.path.basename(
+                     system_info['platform']['description']))[0]
+            board_file = self.conf['board'].lower() + suffix + '.json';
+            self._board_path = os.path.join(basepath, 'libs/utils/platforms',
+                                      board_file)
+            if not os.path.exists(self._board_path):
+                self._log.warning('Board descriptor file %s was not found.',
+                                  self._board_path)
+                self._board_path = None
+                board = None
+            else:
+                board = self._load_board()
```

Comment:
Here we are:
1. refactoring the `_load_board()` call to setup `self._board_path`, which is not required by the existing boards
2. overwriting the anticipated definition of `self._board_path` in the new case of the GEM5 target
3. use the refactored `_load_board()` to actually load a board

I seems to me that all this dance is required just to load a board file which has a suffix.

Cannot we do that by just using the existing `_load_board()` (i.e. no refactoring required) and passing in the exactly `board_file` you build in the first two code line of this chunk?

IOW, cannot this patch be reduced to just 4 lines of code: the three just described here:

```python
suffix = os.path.splitext(os.path.basename(
                system_info['platform']['description']))[0]
                board_file = self.conf['board'].lower() + suffix + '.json';
board = self._load_board(board_file)
```

and the one you add in the following chunk (see my next comment)